client/src/helpers/APICalls/bookingService.ts

```diff
@@ -0,0 +1,74 @@
+import { BookingApiData, BookingRequest } from '../../interface/BookingApiData';
+import { FetchOptions } from '../../interface/FetchOptions';
+import { Modify } from './../../context/useRequestContext';
+
+interface UpdateBooking {
+  accepted?: boolean;
+  declined?: boolean;
+}
+
+export const getBookings = async (): Promise<Array<BookingRequest>> => {
+  const fetchOptions: FetchOptions = {
+    method: 'GET',
+    credentials: 'include',
+  };
+  const res = await fetch('/request', fetchOptions);
+  if (res.status !== 200) {
+    handleError(res);
+    throw new Error();
+  }
+  return await res.json();
+};
+
+export const updateBooking = async (value: Modify, _id: string): Promise<BookingRequest> => {
+  const body: UpdateBooking = {};
+
+  if (value === 'Accept') body.accepted = true;
+  if (value === 'Decline') body.declined = true;
+
+  const fetchOptions: FetchOptions = {
+    method: 'PUT',
+    headers: { 'Content-Type': 'application/json' },
+    credentials: 'include',
+    body: JSON.stringify(body),
+  };
+  const res = await fetch(`/request/${_id}`, fetchOptions);
+  if (res.status !== 200) {
+    handleError(res);
+    throw new Error();
+  }
+  return await res.json();
+};
+
+export function sortBookings(bookings: Array<BookingRequest>): BookingApiData {
+  // sort bookings from Newest to oldest
+  bookings.sort((bookingA, bookingB) => {
+    const startA = new Date(bookingA.start);
+    const startB = new Date(bookingB.start);
+    return startB.getTime() - startA.getTime();
+  });
+
+  const today = Date.now();
+  const past: BookingRequest[] = [];
+
+  const current = bookings.filter((booking) => {
+    const start = new Date(booking.start);
+    if (start.getTime() > today) return true;
+    past.push(booking);
+    return false;
+  });
+
+  const upcomingIdx = current.length - 1;
+  const upcoming = current[upcomingIdx];
+  current.splice(upcomingIdx, 1);
+  return {
+    upcoming,
+    current,
+    past,
+  };
+}
+
+function handleError(res: Response) {
+  // get error message returned from server if any and log it
+  res.json().then((error) => console.log(error));
+}
```

Comment:
Instead of logging the error, it would be better to let the user know that the update failed with a `snackbar`.

Reply:
I agree and I already implemented that logic in the RequestContextProvider component, please check that out in the `useRequesContext.tsx` in the `src/context/` folder. I apologize for the misinformation, I should have stated this earlier.